zombies.js

```diff
@@ -317,14 +330,177 @@
  */
 
 
+ class Item {
+    constructor(name){
+      this.name = name;
+      console.log(name);
+    }
+}
+
+class Weapon extends Item {
+  constructor(name, damage){
+   super(name, damage);
+   this.damage = damage;
+    }
+}
+
+class Food extends Item{
+  constructor(name, energy){
+    super(name, energy);
+    this.energy = energy;
+  }
+}
+
+
+class Player{
+
+  constructor(name, health, strength, speed){
+    this._pack = [];
+    this._maxHealth = health;
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this.equipped = false;
+
+}
+  getPack(){
+    return this._pack;
+  }
+  getMaxHealth(){
+    return this._maxHealth;
+  }
+
+
+
+  takeItem(item){
+    if (this._pack.length < 3){
+      this._pack.push(item);
+      console.log(this.name + " has successfully added " + item.name + " into their bag.");
+      return true;
+    } else {
+      console.log(this.name + " \ 's bag is full. " + item.name + " was not added.");
+      return false;
+    }
+  }
+
+  discardItem(item){
+    let itemLocation = this._pack.indexOf(item);
+
+    if(itemLocation > -1){
+      this._pack.splice(itemLocation, 1);
+      console.log(item.name + " was removed from bag.");
+      return true;
+    } else {
+      console.log(item.name + " not found");
+      return false;
+    }
+
+  }
+
+  checkPack(){
+      console.log(this.getPack());
+  }
+
+  equip(itemToEquip){
+    let itemLocation = this._pack.indexOf(itemToEquip);
+    if(itemToEquip instanceof Weapon && itemLocation > -1 && !this.equipped){
+      this.equipped = itemToEquip;
+      this._pack.splice(itemLocation, 1);
+
+  } else if (this.equipped !== false) {
+      let newWeapon = this.equipped;
+      this.equipped = itemToEquip;
+      this._pack.splice(itemLocation, 1);
+      this._pack.push(newWeapon);
+
+    }
+
+  }
+  eat(itemToEat){
+    let itemLocation = this._pack.indexOf(itemToEat);
+    if(itemToEat instanceof Food && itemLocation > -1 && (this.health + itemToEat.energy) > this.getMaxHealth()){
```

Comment:
Your are doing too much in this conditional. First check if it is Food and in your pack before worrying about maxHealth